### ISSUE TITLE
local_resource with updating status when depends resource without waiting server:ready on kubernetes resource

### ISSUE THREAD

**Issue by drano** [tilt-dev/tilt#4602]
My local_resource starts to update (updating) when it depends on kubernetes resource and this resource is update:success, but it should be started when this resource is server:ready.

**Comment by nicks** [comment]
@drano thanks for the report!

I played around with this a bit and was unable to reproduce the issue you describe. Here's the code that I wrote to try to reproduce: 
https://github.com/tilt-dev/tilt-example-html/commit/5bd074cafb60fd767cc8f1ec2049a1404c1367b9
https://github.com/tilt-dev/tilt-example-html/tree/nicks/issue4602/0-base

I created a pod that sleeps for 15 seconds before it passes the health check. I confirmed that the local_resource ways for the healthcheck to pass before running. Here's a snapshot:
https://cloud.tilt.dev/snapshot/AfjJ8osMBtyHoEh_o4U=

Am I missing something?
